src/manager.rs

```diff
@@ -0,0 +1,74 @@
+use crate::envelope::Envelope;
+use crate::{Actor, Address, Context};
+use futures::channel::mpsc::{self, UnboundedReceiver};
+use futures::StreamExt;
+
+pub struct ActorManager<A: Actor> {
+    receiver: UnboundedReceiver<Box<dyn Envelope<Actor = A>>>,
+    actor: A,
+    ctx: Context<A>,
+}
+
+impl<A: Actor> Drop for ActorManager<A> {
+    fn drop(&mut self) {
+        self.actor.stopped(&mut self.ctx);
+    }
+}
+
+impl<A: Actor> ActorManager<A> {
+    #[cfg(any(feature = "with-tokio-0_2", feature = "with-async_std-1"))]
+    pub(crate) fn spawn(actor: A) -> Address<A> {
+        let (addr, mgr) = Self::start(actor);
+
+        #[cfg(feature = "with-tokio-0_2")]
+        tokio::spawn(mgr.manage());
+
+        #[cfg(feature = "with-async_std-1")]
+        async_std::task::spawn(mgr.manage());
+
+        addr
+    }
+
+    pub(crate) fn start(actor: A) -> (Address<A>, ActorManager<A>) {
+        let (sender, receiver) = mpsc::unbounded();
+        let ctx = Context::new();
+        let mgr = ActorManager {
+            receiver,
+            actor,
+            ctx,
+        };
+        let addr = Address { sender };
+
+        (addr, mgr)
+    }
+
+    //    /// Handle notifications (messages to self)
```

Comment:
These TODOs here too?

Reply:
The bottom TODO applies to this. Didn't want to duplicate it because when I implement notifications again I will know to look here.